### ISSUE TITLE
fullDetail not working in search

### ISSUE THREAD

**Issue by facundoolano** [facundoolano/google-play-scraper#311]
* google-play-scraper version: 6.2.6

### Description: 
After the recent restructuring of the code, the `fullDetail` option of the search method is not being honored.

We need to implement something similar as done in parseList, where a new app method call is done per each yielded result.

Also add a test proving that full detail works for search.

### Example code:
```js
g.search({term: 'panda', num:1, fullDetail: true }).then(console.log)

[ { title: 'Panda Pop! Bubble Shooter Saga & Puzzle Adventure',
    appId: 'com.sgn.pandapop.gp',
    url: 'https://play.google.com/store/apps/details?id=com.sgn.pandapop.gp',
    icon: 'https://lh3.googleusercontent.com/n-N5SeSksOVootPWoYDGek1iY-_EeoL46hhWOgnxbbP0eypVMqMlJtOwAZSzN-yDoOtO',
    developer: 'Jam City, Inc.',
    developerId: '5509190841173705883',
    priceText: 'FREE',
    free: true,
    summary: 'Aim, match, shoot and pop in this free bubble shooter classic saga adventure!',
    scoreText: '4.4',
    score: 4.4345155 } ]
```

**Comment by v1shky** [comment]
Any progress on this issue by any forks?

**Comment by efeconirulez** [comment]
+1

**Comment by girotomas** [comment]
```
gplay
  .search({
    throttle: 5,
    term: "panda",
    num: 2,
    fullDetail: true
  })
  .then(
    l =>
      l.map(a => gplay.app({ appId: a.appId }).then(console.log, console.log)),
    console.log
  );
```

**Comment by efeconirulez** [comment]
Still no progress?

**Comment by sgallese** [comment]
fullDetail seems to be working for me.

**Comment by KavarH** [comment]
For me it is still not working.
It is just showing a short summary.
Doing what @girotomas wrote you see the difference.

**Comment by eltriplet** [comment]
> This might be useful for someone
> 
> ```
> gplay
>   .search({
>     throttle: 5,
>     term: "panda",
>     num: 2,
>     fullDetail: true
>   })
>   .then(
>     l =>
>       l.map(a => gplay.app({ appId: a.appId }).then(console.log, console.log)),
>     console.log
>   );
> ```

If you want to store the result in a json array, instead of printing on terminal, the following adaptation of @girotomas code worked for me:

var gplay = require('google-play-scraper');

justDoIt();

async function justDoIt() {
    let res = await gplay.search({
        throttle: 5,
        term: "panda",
        num: 2,
        fullDetail: true
    })
    res = await getFullRes(res);
    console.log(res);
}

async function getFullRes(shortRes) {
    return Promise.all(
        shortRes.map(async a => {
        return await gplay.app({ appId: a.appId })
        })
    )
}

**Comment by KavarH** [comment]
I did start in my fork to work on a fix.

Not sure if its already working in all cases.
Additionally, I'm not such a js pro...
However, maybe something comming soon

**Comment by KavarH** [comment]
Fixed in #384
